src/overview/repository.d

```diff
@@ -120,36 +125,50 @@ void updateRepositoryDependencies ( ref Repository project, HTTPConnection clien
         import std.regex;
         import std.exception : enforce;
 
-        auto info = repo.download(submodule).expectSubmodule();
-
-        // Matching submodule URL is not strictly necessary because all
-        // tagged SHA for all repos are already known (and can be assumed
-        // unique across repos). But doing so helps with early error detection
-        // in case external submodule is used
-
-        static rgxSubmoduleURL = regex(
-            r"(https:\/\/)|(git@)github\.com\/|:([^\/]+)\/([^.]+)\.git");
-
-        auto match = info.url.matchFirst(rgxSubmoduleURL);
-        enforce(
-            !match.empty,
-            format("Unexpected submodule URL (%s) format", info.url)
-        );
-
-        auto p_version = info.sha in mapping;
-        enforce(
-            p_version !is null,
-            format("SHA '%s' for '%s' not found in mapping", info.sha, info.url)
-        );
-
-        return tuple(match[3] ~ "/" ~ match[4], *p_version);
+        try
+        {
+            auto info = repo.download(submodule).expectSubmodule();
+
+            // Matching submodule URL is not strictly necessary because all
+            // tagged SHA for all repos are already known (and can be assumed
+            // unique across repos). But doing so helps with early error detection
+            // in case external submodule is used
+
+            static rgxSubmoduleURL = regex(
+                r"(https:\/\/)|(git@)github\.com\/|:([^\/]+)\/([^.]+)\.git");
+
+            auto match = info.url.matchFirst(rgxSubmoduleURL);
+            enforce(
+                !match.empty,
+                format("Unexpected submodule URL (%s) format", info.url)
+            );
+
+            auto p_version = info.sha in mapping;
+            enforce(
+                p_version !is null,
+                format("SHA '%s' for '%s' not found in mapping", info.sha, info.url)
+            );
+
+            return tuple(match[3] ~ "/" ~ match[4], *p_version);
+        }
+        catch (Exception)
+            return tuple("", Version.init);
     }
 
-    alias submodules = repo => repo.download("submodules/").expectDirectory.listAll();
+    try
+    {
+        alias submodules = repo => repo.download("submodules/").expectDirectory.listAll();
```

Comment:
You don't need a git repo to use `git config -f <file>`. But I guess the git config ini format is simple enough to parse, is just that I prefer to rely on tools in case there is any changes in formats.